src/core/utils.js

```diff
@@ -137,6 +136,10 @@ export function isAllowedOrigin(urlString, {
   return allowedOrigins.includes(`${parsedURL.protocol}//${parsedURL.host}`);
 }
 
+export function getIconUrl(addon, fallback) {
+  return (addon && isAllowedOrigin(addon.icon_url)) ? addon.icon_url : fallback;
```

Comment:
I think it would be nice to make the `fallback` parameter be optional. By default it can use `'amo/img/icons/default-64.png'` like the original code did.